src/app/main.rs

```diff
@@ -574,6 +572,9 @@ where
                         })
                         .ok()
                 }))
+                // `limit` must be shared to apply to all connections...
+                .push(limit::layer(max_in_flight))
```

Comment:
If I understand what is meant by applying to all connections, wouldn't this *outer* limit layer load shed before any of the routers below it since they are all the same value? Is it redundant to have limit layers in any of the routers above this?

Reply:
> wouldn't this outer limit layer load shed before any of the routers below it since they are all the same value?

Yes, that's the goal, since the routers can siphon off requests in different buckets, but we want the *total* limit.